plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4623,6 +4623,48 @@ public void testPredicatePushDownToTableScan()
         }
     }
 
+    @Test
+    public void testMismatchedBucketWithBucketPredicate()
+    {
+        try {
+            assertUpdate(
+                    "CREATE TABLE test_mismatch_bucketing8\n" +
+                            "WITH (bucket_count = 8, bucketed_by = ARRAY['key8']) AS\n" +
+                            "SELECT custkey key8, comment value8 FROM orders",
+                    15000);
+            assertUpdate(
+                    "CREATE TABLE test_mismatch_bucketing32\n" +
+                            "WITH (bucket_count = 32, bucketed_by = ARRAY['key32']) AS\n" +
+                            "SELECT custkey key32, comment value32 FROM orders",
```

Comment:
Is this testable with smaller tables?
Like `nation` with bucketing on `nationkey`?

Reply:
Thanks!